operator/api/v1alpha1/plan_types.go

```diff
@@ -20,27 +20,64 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
-// PlanSpec defines the desired state of Plan
+// PlanSpec defines the Service Plan spec.
 type PlanSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// The name of the Offering this Plan belongs to.
+	Offering string `json:"offering,omitempty"`
+
+	// A unique ID for the Plan.
+	ID string `json:"id,omitempty"`
+	// A description for the Plan.
+	Description string `json:"description,omitempty"`
+
+	Provisioning PlanProvisioningSpec `json:"provisioning,omitempty"`
+	Binding      PlanBindingSpec      `json:"binding,omitempty"`
+}
+
+// PlanProvisioningSpec defines the Provisioning spec for a Plan.
+type PlanProvisioningSpec struct {
+	Chart  PlanProvisioningChartSpec  `json:"chart,omitempty"`
+	Values PlanProvisioningValuesSpec `json:"values,omitempty"`
+}
+
+// PlanProvisioningChartSpec defines the Chart spec for a Provisioning.
+type PlanProvisioningChartSpec struct {
+	URL string `json:"url,omitempty"`
+}
+
+// PlanProvisioningValuesSpec defines the Values spec for a Provisioning. It includes the JSON
+// schema for validation, the default values that can be overriden by the user, and the static
+// values that enforce plan-specific configuration.
+type PlanProvisioningValuesSpec struct {
+	Schema  string `json:"schema,omitempty"`
+	Default string `json:"default,omitempty"`
+	Static  string `json:"static,omitempty"`
+}
+
+// PlanBindingSpec defines the Binding spec for a Plan.
```

Comment:
This needs more useful comments, like what a binding is.

Reply:
Done with https://github.com/SUSE/metabroker/pull/6/commits/af94989eecc3428dc43509716746d5610a2994a8.